tools/concrete-protocol/src/concrete-protocol.proto

```diff
@@ -0,0 +1,295 @@
+syntax = "proto3";
+
+package concreteprotocol;
+
+/////////////////////////////////////////////////////////////////////////////////////////// Commons
+
+enum KeyType{
+    binary = 0;
+    ternary = 1;
+}
+
+message NativeModulus{}
+
+message PowerOfTwoModulus{
+    uint32 power = 1;
+}
+
+message IntegerModulus{
+    uint32 modulus = 1;
+}
+
+message Modulus {
+    oneof modulus {
```

Comment:
Why are you separate this modulus, why they can't be just an integer modulus?

Reply:
If you want to use the native modulus, for say 64 bits integers, you would have to write 2**64. That would take 65 bits which can´t be done with the protobuf types. Also, that would be a waste of space. Going further, since we at least need the `NativeModulus` and the `IntegerModulus`, I thought we could as well provide a `PowerOfTwoModulus` that allows to reduce the size.